pkg/health/server.go

```diff
@@ -58,35 +60,32 @@ func (s *server) Run(ctx context.Context, port int) error {
 
 	//nolint:gosec
 	srv := &http.Server{
-		Addr:    fmt.Sprintf(":%d", port),
-		Handler: router,
+		Addr:        fmt.Sprintf(":%d", port),
+		Handler:     router,
+		BaseContext: func(_ net.Listener) context.Context { return ctx },
 	}
 
+	serveErr := make(chan error, 1)
```

Comment:
I'm not sure I understand this change.

You're starting the server in background and capturing the error in serveErr. However, you're also blocking until the context is canceled.
So in order to be able to return the error, the context has to be canceled. Unlike the existing version, if the server fails to start there isn't even a log now.

Reply:
Thanks for the catch! This should be:


```go
	select {
	case <-serveErr:
		return serveErr
	case <-ctx.Done():
	}
```

instead of just

```go
<-ctx.Done()
```